app/views/landing_page/blocks/_box_container.html.erb

```diff
@@ -0,0 +1,16 @@
+<%
+  data_attributes = {
+    module: "ga4-link-tracker",
+    ga4_set_indexes: "",
+    ga4_track_links_only: "",
+    ga4_link: {
+      event_name: "navigation",
+      type: "box",
+    }.to_json
+  }
+%>
+<%= tag.div data: data_attributes do %>
+  <% block.blocks.each do |child| %>
```

Comment:
Instead of creating a new block and wrapping the whole page in it, could these tracking attributes be applied to the show template? https://github.com/alphagov/frontend/blob/main/app/views/landing_page/show.html.erb#L25

Reply:
Ah no this is different - the box container I've created is only wrapped around all the block we called "box", not the whole page:

<img width="1068" alt="image" src="https://github.com/user-attachments/assets/23342ee9-a1e1-4f5f-939e-6c44dfe9d27e" />

Reply:
My concern with this approach is that we have to add an extra layer of blocks just for tracking, which seems like it could be over complicating a publishing interface. Did you consider other options? Perhaps as an option on the columns_layout block?

Reply:
Happy to move it to the columns layout 👍  - I think I was worried that adding tracking to a "layout" block would be confusing from an architectural perspective, but I think what you said makes sense. We'll need to pass `ga4_tracking: true` and `ga4_type: box` through on the column layout we want to track so that it doesn't affect anything else using the column layout, and so the tracking isn't tied to boxes specifically.

Reply:
I think that sounds sensible. It would be great if it could automatically figure out the kind of tracking it needs based on the blocks themselves (and likewise for any future tracking, so there's just one parameter to turn it on) but don't worry if that's too complex.

Reply:
Thanks, yeah that would be good - though sometimes the PAs use a different name to what we've called the block, which might complicate things!

Reply:
Fair enough, what you've done now looks good 👍